modules/renter/hostdb/hostweight.go

```diff
@@ -322,13 +322,9 @@ func (hdb *HostDB) uptimeAdjustments(entry modules.HostDBEntry) float64 {
 			// Ignore the unsorted scan entry.
 			continue
 		}
-		if recentSuccess {
-			uptime += scan.Timestamp.Sub(recentTime)
-		} else {
-			downtime += scan.Timestamp.Sub(recentTime)
-		}
-		recentTime = scan.Timestamp
-		recentSuccess = scan.Success
+
+		decayUptimeOrDowntime(&entry, scan, recentScan)
```

Comment:
This is the wrong place to apply decay - you are applying decay every time that the weight is being calculated. But we can calculate the weight for a lot of different reasons. We should only be applying decay after we scan the host. That means the decay algorithm should also be moved to `scan.go`.